main.go

```diff
@@ -1,20 +1,58 @@
 package main
 
 import (
+	"context"
 	"io/ioutil"
 	"log"
 	"net/http"
 	"os"
 	"time"
 
 	"github.com/go-git/go-git/v5"
+	"github.com/google/go-github/v47/github"
 	"github.com/julienschmidt/httprouter"
 )
 
 const repoURL = "https://github.com/RedHatInsights/insights-core"
 
 var repoPath = ""
 
+var forksCache = make([]map[string]string, 0)
+
+var forkCacheTimestamp = time.Now()
+
+func getGithubForks() error {
+	var err error
+	forksCache = nil
```

Comment:
Because `getGithubForks` runs on a separate goroutine from the HTTP request handlers, setting the global variable `forksCache` to `nil` before a new slice is ready creates the possibility that the `getForksCache` HTTP request handler will attempt to marshal `forksCache` while it is `nil`. To prevent this, you could do one of two things.

1) Don't assign it the value `nil`; just assign it to the new `forks` slice when that slice is ready.
2) Guard the assignment of `forksCache = forks` with a `sync.RWMutex` lock to block an accessing goroutine from reading `forksCache` while it is being assigned to.

Reply:
Oh that's right ( I forget about the different goroutines). I have added the RWMutex to block all the access to the list `forkCache`